exec/plugin-commands-script-runners/src/runRecursive.ts

```diff
@@ -57,12 +58,19 @@ export async function runRecursive (
   const existsPnp = existsInDir.bind(null, '.pnp.cjs')
   const workspacePnpPath = opts.workspaceDir && await existsPnp(opts.workspaceDir)
 
+  const requiredScripts = opts.rootProjectManifest?.pnpm?.requiredScripts ?? []
+
   for (const chunk of packageChunks) {
     await Promise.all(chunk.map(async (prefix: string) =>
       limitRun(async () => {
         const pkg = opts.selectedProjectsGraph[prefix]
+        if (!pkg.package.manifest.scripts?.[scriptName]) {
+          if (requiredScripts.includes(scriptName)) {
+            throw new PnpmError('RECURSIVE_RUN_NO_SCRIPT', `Missing script "${scriptName}" in ${pkg.package.dir}`)
```

Comment:
One observation is that this throws an error immediately in a async function, which means

1. Non deterministic order 
2. Bad user experience to fix the errors

Let's say projects A, B are missing the target scripts:

1. people may get missing script error for A, sometimes get missing script error for B.
2. when a user get a missing script error for A, he/she fixes it and then run the same command again. He/She may encounter a missing script error for another project. This kind of experience feels a bit endless...

So, IMO, it will be better to print something like `Missing script "${scriptName}" in packages: @my-company/A, @my-company/B`. In this way, people can get all the projects miss the target script in one operation.